packages/lib/apps/getTeamAppCredentials.ts

```diff
@@ -0,0 +1,27 @@
+import type { Prisma } from "@prisma/client";
+
+import type { TeamQuery } from "@calcom/ee/teams/teams.repository";
+import type { CredentialPayload } from "@calcom/types/Credential";
+
+export type Credentials = {
+  user: {
+    email: string;
+  } | null;
+  id: number;
+  userId: number | null;
+  type: string;
+  key: Prisma.JsonValue;
+  teamId: number | null;
+  appId: string | null;
+  invalid: boolean | null;
+}[];
+
+const getTeamAppCredentials = (userTeams: TeamQuery[]) => {
+  const teamAppCredentials: CredentialPayload[] = userTeams.flatMap((teamApp) => {
+    return teamApp.credentials ? teamApp.credentials.flat() : [];
+  });
+
+  return teamAppCredentials;
+};
```

Comment:
```suggestion
const getTeamAppCredentials = (userTeams: TeamQuery[]): CredentialPayload[]  => {
  return userTeams.flatMap((teamApp) => teamApp.credentials ? teamApp.credentials.flat() : []);
};
```

Constant assignment is not being used anywhere else